core/domain/user_domain.py

```diff
@@ -797,6 +797,31 @@ def remove_learnt_topic_id(self, learnt_topic_id):
         self.learnt_topic_ids.remove(learnt_topic_id)
 
 
+class LearnerGoals(python_utils.OBJECT):
+    """Domain object for the learner goals model."""
+
+    def __init__(
+            self, user_id, topic_ids_to_learn):
+        self.id = user_id
+        self.topic_ids_to_learn = topic_ids_to_learn
+
+    def add_topic_id_to_learn(self, topic_id):
+        """Adds the topic id to learn list.
+
+        Args:
+            topic_id: str. The topic id to be inserted to the learn
+                list.
+        """
+        self.topic_ids_to_learn.append(topic_id)
```

Comment:
Do you need to check for duplicates or is this validated somewhere pre-save?

Reply:
Before adding the topic_id to the list we check if the topic_id is not already present in the list (in the learner_progress_services.add_topic_to_learn()).